server/chat.ts

```diff
@@ -1141,6 +1116,55 @@ export class CommandContext extends MessageContext {
 
 		return message;
 	}
+	canPM(targetUser: User) {
+		// hotpatching
+		const user = this.user;
+		if (typeof user.settings.blockPMs === 'boolean') {
+			user.settings.blockPMs = {all: user.settings.blockPMs};
+		}
+		if (typeof targetUser.settings.blockPMs === 'boolean') {
+			targetUser.settings.blockPMs = {all: targetUser.settings.blockPMs};
+		}
+
+		function checkBlocked(curUser: User, tarUser: User) {
+			const curBlocks = curUser.settings.blockPMs;
+			const tarBlocks = tarUser.settings.blockPMs;
+
+			return (tarBlocks.all === true ||
+				!Users.globalAuth.atLeast(user, tarBlocks.all as GroupSymbol) && typeof curBlocks.all === 'string' ||
+				tarBlocks.specific?.includes(curUser.id)
+			) && !curUser.can('lock');
+		}
+
+		if (targetUser.locked && !user.can('lock')) {
+			this.errorReply(`The user "${targetUser.name}" is locked and cannot be PMed.`);
+			return null;
+		}
+		if (Config.pmmodchat && !Users.globalAuth.atLeast(user, Config.pmmodchat) &&
+			!Users.Auth.hasPermission(targetUser, 'promote', Config.pmmodchat as GroupSymbol)
```

Comment:
```suggestion
		if (
			Config.pmmodchat && !Users.globalAuth.atLeast(user, Config.pmmodchat) &&
			!Users.Auth.hasPermission(targetUser, 'promote', Config.pmmodchat as GroupSymbol)
```

Reply:
Still really sad eslint can't enforce this.

Reply:
There's `array-bracket-newline`, `function-paren-newline`, and `object-curly-newline`, but no generalized paren newline. :[